### ISSUE TITLE
Regex `set` + toObject options causing Cast to Embedded error

### ISSUE THREAD

**Issue by wlingke** [Automattic/mongoose#5601]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**

This script below causes an error: `ValidationError: Parent validation failed: child: Cast to Embedded failed for value "{ number: '55555555-5555' }" at path "child"`
```
const mongoose = require('mongoose');

mongoose.connect('mongodb://127.0.0.1/test');
const db = mongoose.connection;

db.once('open', async () => {
  try {
    const ChildSchema = new mongoose.Schema({
      number: {
        type: String,
        set: v => v.replace(/\D/g, ''),
      },
      _id: false,
    });
    ChildSchema.set('toObject', { getters: true, minimize: false });

    const ParentSchema = new mongoose.Schema({
      child: {
        type: ChildSchema,
        default: {},
      },
    });

    const Parent = mongoose.model('Parent', ParentSchema);

    const ogParent = new Parent();
    ogParent.child = { number: '55555555-5555' };
    await ogParent.save();

    const foundParent = await Parent.findById(ogParent._id).exec();
    console.log(foundParent);

    db.close();
  } catch (e) {
    console.log(e);
  }
});

```

Odd behavior includes if I made set `set: v=> v+"_"` instead of that regex, it works. If I remove {getters: true} it works. If I remove {minimize: true} it works.

Not really sure why this is happening... 

Adding a `default: ''` to `number` does seem to solve this problem.

~~This is an issue with latest 4.11.9 but NOT 4.11.8~~

**Comment by wlingke** [comment]
So after additional investigation, it looks like this is because the `set` is getting called TWICE. First with `undefined` then with the string. So `undefined.replace()` causes an error which just ends up propogating out as a `cast embed` error.

The solution is to just check for `undefined` in the setter then.

Aside from that, Is it normal for the setter to be called twice?

**Comment by varunjayaraman** [comment]
thanks for the repro script, this does look like a bug

**Comment by vkarpov15** [comment]
Fix will be in for 4.11.11 :+1:
